### ISSUE TITLE
Caused by: java.lang.IllegalStateException: This build is not parameterized!

### ISSUE THREAD

**Issue by kargakis** [openshift/release#235]
Seeing this error in the new jenkins master
```
Caused by: java.lang.IllegalStateException: This build is not parameterized!
	at jenkins.model.ParameterizedJobMixIn.doBuildWithParameters(ParameterizedJobMixIn.java:238)
	at org.jenkinsci.plugins.workflow.job.WorkflowJob.doBuildWithParameters(WorkflowJob.java:290)
	at java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:627)
	at org.kohsuke.stapler.Function$MethodFunction.invoke(Function.java:343)
	... 88 more
Oct 05, 2017 10:48:57 PM org.eclipse.jetty.server.handler.ContextHandler$Context log
WARNING: Error while serving https://jenkins-origin-ci.svc.ci.openshift.org/job/origin-ci-verify-origin/buildWithParameters
```
Seems like parameters are removed from all jobs.

/assign @csrwng

**Comment by kargakis** [comment]
cc @stevekuznetsov

**Comment by csrwng** [comment]
@kargakis so I naively went and updated the memory limit for the Jenkins server which caused a redeploy. Because we're not using a pv, all jenkins data was blown away. The jobs are now recreated brand new by the jjb controller, but previous run data is gone.

Because Jenkins hasn't run a pipeline job, it doesn't know what parameters it takes. One option to fix this for new jobs is just to specify the parameters through the jjb yaml. Only after the first run does it realize that it has parameters.

**Comment by kargakis** [comment]
Opened https://github.com/openshift/release/issues/239 to track persisting the new jenkins data.

**Comment by kargakis** [comment]
I had to manually start the pipelines for the parameters to be picked up. There were other failed jobs already so I am not sure why the parameters weren't there in the first place. I also had to rekick a bunch of PR jobs that were broken because of this. Not the best experience. We need to automatically get the parameters from start.

**Comment by csrwng** [comment]
Agreed. If I add them to the jjb yaml definition, they should come up from the beginning. Will test that today.

**Comment by csrwng** [comment]
Fixed with #236
